hubspot-style/src/test/java/com/hubspot/immutables/utils/WireSafeEnumTest.java

```diff
@@ -392,4 +406,24 @@ public void itDoesNotThrowForValueFromKnownString() {
     assertThat(wrapper.asEnumOrThrow())
         .isEqualTo(RetentionPolicy.SOURCE);
   }
+
+  @Test
+  public void itDelegatesToJsonCreatorIfNotCached() throws Exception {
+    WireSafeEnum<EnumWithMultipleSerializedForms> wrapper = MAPPER.readValue(
+        "\"ABC\"",
+        new TypeReference<WireSafeEnum<EnumWithMultipleSerializedForms>>() {}
+    );
+    assertThat(wrapper.enumType()).isEqualTo(EnumWithMultipleSerializedForms.class);
+    assertThat(wrapper.asString()).isEqualTo("ABC");
+    assertThat(wrapper.asEnum()).contains(EnumWithMultipleSerializedForms.ABC);
+
+    wrapper = MAPPER.readValue(
+        "\"abc\"",
+        new TypeReference<WireSafeEnum<EnumWithMultipleSerializedForms>>() {}
+    );
+    assertThat(wrapper.enumType()).isEqualTo(EnumWithMultipleSerializedForms.class);
+    assertThat(wrapper.asString()).isEqualTo("abc");
+    assertThat(wrapper.asEnum()).contains(EnumWithMultipleSerializedForms.ABC);
+  }
```

Comment:
Should be a 3rd case here for something like "xyz"? I assume it's mostly covered by other tests but seems like it would be good to have here for completeness